src/server/tiering/common.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2024, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <memory>
+#include <optional>
+
+#include "util/fibers/synchronization.h"
+
+namespace dfly::tiering {
+
+// Location on the offloaded blob, measured in bytes
+struct DiskSegment {
+  DiskSegment() = default;
+  DiskSegment(size_t offset, size_t length) : offset{offset}, length{length} {
+  }
+  DiskSegment(std::pair<size_t, size_t> p) : offset{p.first}, length(p.second) {
+  }
+
+  size_t offset = 0, length = 0;
+};
+
+// Simple thread-safe fiber-blocking future for waiting for value. Pass by value.
+template <typename T> struct Future {
```

Comment:
why not use fb2::Future<> / fb2::Promise<> ?

Reply:
I forgot we have it at all, we use it only once in DF + helio. The internals also look frightening but I'll replace it 😆